### ISSUE TITLE
How to deploy to a virtual Directory using GitHub action "azure/webapps-deploy"

### ISSUE THREAD

**Issue by jlara005** [Azure/webapps-deploy#340]
Is there a way to deploy to a virtual Directory using GitHub action "azure/webapps-deploy"? I don't see the equivalent in the GitHub action.

In Azure DevOps you were able to do this:
![image](https://github.com/Azure/webapps-deploy/assets/29823126/eb318aec-c9be-4e89-8297-79bca52af9e4)

**Comment by va-timorris** [comment]
Hi NoCopy - this did not work for us.

We have one Azure App Service which uses /wwwroot and a Virtual Directory on that App Service that uses /api

We use two Github workflows.  One for the web app (/wwwroot) and another one for the API (/api).

What happens is that we can deploy to the virtual directory (/api) in its workflow, but then when we run the web app workflow, it also deploys to the /api folder.

The only way I've gotten this to work is by manually editing the following file in Kudu:  D:\home\site\deployments\tools\deploy.cmd

In that file, I set the DEPLOYMENT_TARGET based on the existence of a DLL in the DEPLOYMENT_SOURCE.  If the DEPLOYMENT_SOURCE contains the DLL associated with the API, I set the DEPLOYMENT_TARGET to /api.  Else if it contains the DLL associated with the Web App, I set DEPLOYMENT_TARGET to /wwwroot

The problem  with this approach is that the 'deploy.cmd' file randomly gets reset and I lose my changes to it!  I have no idea what is resetting this file.

Below is the relevant YAML for our deployments:


API workflow file
-----------------------------------------------------------------
env:
  AZURE_DEPLOYMENT_TARGET: 'D:\home\site\api'

    - name: Configure Azure Deployment Target
      uses: Azure/login@v1.4.6
      with:
          creds: ${{ secrets.ATC_DEVTEST_EXT_GITHUB_SP }}
          environment: 'AzureUSGovernment'
          run: az webapp config appsettings set --name ${{ secrets.ATC_APP_NAME }} --resource-group ${{ env.AZURE_RESOURCE_GROUP }} --settings DEPLOYMENT_TARGET=${{ env.AZURE_DEPLOYMENT_TARGET }} --output none

    - name: 'Deploy to Azure App Service using DEV publish profile credentials'
      uses: azure/webapps-deploy@v2.2.10
      with:
        app-name: ${{ secrets.ATC_APP_NAME }}
        publish-profile: ${{ secrets.ATC_PUBLISH_PROFILE_API }}
        package: '.'


Web App workflow file
-----------------------------------------------------------------
env:
  AZURE_DEPLOYMENT_TARGET: 'D:\home\site\wwwroot'

    - name: Configure Azure Deployment Target
      uses: Azure/login@v1.4.6
      with:
          creds: ${{ secrets.ATC_DEVTEST_EXT_GITHUB_SP }}
          environment: 'AzureUSGovernment'
          run: az webapp config appsettings set --name ${{ secrets.ATC_APP_NAME }} --resource-group ${{ env.AZURE_RESOURCE_GROUP }} --settings DEPLOYMENT_TARGET=${{ env.AZURE_DEPLOYMENT_TARGET }} --output none

    - name: 'Deploy to Azure App Service using DEV publish profile credentials'
      uses: azure/webapps-deploy@v2.2.10
      with:
        app-name: ${{ secrets.ATC_APP_NAME }}
        publish-profile: ${{ secrets.ATC_PUBLISH_PROFILE_API }}
        package: '.'

**Comment by NoCopy** [comment]
@va-timorris well your AzureUSGovernment environment is set on both actions so obviously your env.AZURE_DEPLOYMENT_TARGET is going to be the same. 

Either use different vars, or different environments.

**Comment by va-timorris** [comment]
@NoCopyright

I was under the impression that environment: 'AzureUSGovernment' was just applicable to the Azure Environment I'm logging into.   I don't think it as to do with Github environment.  We don't have a Github environment named 'AzureUSGovernment'.

Also, in your sample code, you're using ${{ vars.AZURE_WEBAPP_NAME }}

Where is this defined?   Is there a section in the workflow yaml file like:

vars:
  AZURE_WEBAPP_NAME: '...'

**Comment by NoCopy** [comment]
@va-timorris sorry, I might have misunderstood that. 

Github does have the concept of different environments. You can go your repo > settings > environments and then specify specific secrets / vars for your specific ENV. I didn't realize this plugin has an environment setting as well. 

Anyways, I am guessing you are just using the standard repo vars; which is fine, but like I said, I couldn't get it to work with the env: line.

Try this:

env.AZURE_DEPLOYMENT_TARGET  = wwwroot
env.AZURE_DEPLOYMENT_API_TARGET = api

That should get you there.

**Comment by NoCopy** [comment]
> for the virtual path modify the zip deploy destination url in ur publish profile

Can't, see #154 

Otherwise this wouldn't be an issue.

You can modify the deploy.cmd directly on the server to change the path but when you want to deploy to a virtual directory it gains you nothing. Hence the request to fix this issue. 

the only other way is what I posted, a run time modification to the deployment path when you are deploying.

**Comment by NoCopy** [comment]
@m1rl0k Right, sorry, there may be some mixup here. There is no direct AZURE_DEPLOYMENT_API_TARGET on the Azure CLI. There isn't even a AZURE_DEPLOYMENT_TARGET for that matter. Those are just the set vars on the repo. I was assuming that the poster was sharing the repo with the same Vars and thus deploying to the same location because I know changing the deploy target works - as per your experience. yet @va-timorris is saying it doesn't, there must be a difference in the configuration somewhere. @va-timorris if you want help walking through it I certainly can. DM me.

**Comment by va-timorris** [comment]
@m1rl0k : in the API workflow, you're piping the az webapp config command like this:    > NUL

But in the frontend workflow, you're piping the az webapp config command lke this:  > dev/null


Is there a reason for this?  I've removed piping that command on the end, I just run the command without that stuff on the end.

**Comment by pramodmd** [comment]
I have the same problem using azure/webapps-deploy@v3. 
Perfectly works for default app hosted at wwwroot. Workflow for wwwrooot/app2 fails.
Can't pass virtual-application as input. git action fails with invalid input.  
Anyone successfully implemented this?

**Comment by m1rl0k** [comment]
> @m1rl0k : in the API workflow, you're piping the az webapp config command like this:    > NUL
> 
> 
> 
> But in the frontend workflow, you're piping the az webapp config command like this:  > dev/null
> 
> 
> 
> 
> 
> Is there a reason for this?  I've removed piping that command on the end, I just run the command without that stuff on the end.

this is because one is windows runner and one is ubuntu

**Comment by Uriil** [comment]
Please look at "target-path" property, it did the trick for me: https://github.com/Azure/webapps-deploy/blob/releases/v3/action.yml
